near-sdk-macros/src/core_impl/info_extractor/attr_sig_info.rs

```diff
@@ -38,25 +38,17 @@ impl AttrSigInfo {
         original_attrs: &mut Vec<Attribute>,
         original_sig: &mut Signature,
     ) -> syn::Result<Self> {
-        if original_sig.asyncness.is_some() {
+        if original_sig
+            .generics
+            .params
+            .iter()
+            .any(|g| matches!(g, GenericParam::Type(_) | GenericParam::Const(_)))
+        {
             return Err(Error::new(
-                original_sig.span(),
-                "Contract API is not allowed to be async.",
+                original_sig.generics.span(),
+                "Contract API is not allowed to have generics.",
```

Comment:
What do you think about reporting errors on a case-by-case basis, so it highlights the generic itself and not the `<` span?

We can use [`syn::Error::combine`](https://docs.rs/syn/latest/syn/parse/struct.Error.html#method.combine) to combine multiple errors, so it highlights all of them. ([example](https://github.com/near/near-sdk-rs/blob/0075253c6d4b862817a8a7fe5105ac87dc223614/near-sdk-macros/src/lib.rs#L368-L370))

And since we've decided to leave generics out, it only highlights the generics but not the lifetimes.